File: courses/cs50x/psets/1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_change(void);
int get_coins(int change);

int main(void)
{
	printf("%d\n", get_coins(get_change()));
	return 0;
}

int get_change(void)
{
	int n;
	do
		n = get_int("Change owed: ");
	while (n < 0);
	return n;
}

int get_coins(int ch)
{
	const int QUARTER = 25;
	const int DIME = 10;
	const int NICKEL = 5;

	int c = 0;
	while (ch > 0)
	{
		if (ch >= QUARTER)
			ch -= QUARTER;
		else if (ch >= DIME)
			ch -= DIME;
		else if (ch >= NICKEL)
			ch -= NICKEL;
		else
			ch--;

		c++;
	}
	return c;
}

File: courses/cs50x/psets/6/sentimental-cash/cash.py
from cs50 import get_float


def main():
    print(get_coins(get_change()))


def get_change() -> int:
    while True:
        change: float | None = get_float("Change: ")
        if change is not None and change >= 0.0:
            return round(change * 100)


def get_coins(change: int) -> int:
    QUARTER = 25
    DIME = 10
    NICKEL = 5
    PENNY = 1

    coins = 0
    while change > 0:
        if change >= QUARTER:
            change -= QUARTER
        elif change >= DIME:
            change -= DIME
        elif change >= NICKEL:
            change -= NICKEL
        else:
            change -= PENNY
        coins += 1
    return coins


main()
